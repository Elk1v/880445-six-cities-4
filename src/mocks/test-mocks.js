import moment from "moment";

export const Options = {
  OFFERS_COUNT: 312,
};

export const offers = [
  {
    id: 0,
    title: `Beautiful & luxurious apartment at great location`,
    features: {
      type: `Apartment`,
      bedrooms: 3,
      adults: 4,
    },
    rating: 4.5,
    isPremium: true,
    price: 100,
    images: [
      `img/studio-01.jpg`,
      `img/apartment-01.jpg`,
      `img/apartment-02.jpg`,
      `img/apartment-03.jpg`,
      `img/studio-01.jpg`,
      `img/apartment-01.jpg`
    ],
    inside: [
      `Wi-Fi`,
      `Washing machine`,
      `Towels`,
      `Heating`,
      `Coffee machine`,
      `Baby seat`,
      `Kitchen`,
      `Dishwasher`,
      `Cabel TV`,
      `Fridge`,
    ],
    host: {
      name: `Angelina`,
      avatar: `img/avatar-angelina.jpg`,
      text: [
        `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam. The building is green and from 18th century.`,
        `An independent House, strategically located between Rembrand Square and National Opera, but where the bustle of the city comes to rest in this alley flowery and colorful.`,
      ]
    },

    coords: [52.3909553943508, 4.85309666406198],

    reviews: [{
      id: 0,
      user: {
        name: `Max`,
        avatar: `img/avatar-max.jpg`
      },
      rating: 4.5,
      text: [
        `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam. The building is green and from 18th century.`,
        `An independent House, strategically located between Rembrand Square and National Opera, but where the bustle of the city comes to rest in this alley flowery and colorful.`,
      ],
      date: moment().format(`YYYY-MM-DD`),
    }],
  },

  {
    id: 1,
    title: `Beautiful & luxurious apartment at great location`,
    features: {
      type: `Apartment`,
      bedrooms: 3,
      adults: 4,
    },
    rating: 4.0,
    isPremium: false,
    price: 70,
    images: [
      `img/studio-01.jpg`,
      `img/apartment-01.jpg`,
      `img/apartment-02.jpg`,
      `img/apartment-03.jpg`,
      `img/studio-01.jpg`,
      `img/apartment-01.jpg`
    ],
    inside: [
      `Wi-Fi`,
      `Towels`,
      `Coffee machine`,
      `Kitchen`,
      `Dishwasher`,
      `Fridge`,
    ],
    host: {
      name: `Max`,
      avatar: `img/avatar-max.jpg`,
      text: [
        `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam. The building is green and from 18th century.`,
        `An independent House, strategically located between Rembrand Square and National Opera, but where the bustle of the city comes to rest in this alley flowery and colorful.`,
      ]
    },

    coords: [52.369553943508, 4.85309666406198],

    reviews: [{
      id: 1,
      user: {
        name: `Max`,
        avatar: `img/avatar-max.jpg`
      },
      rating: 4.5,
      text: [
        `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam. The building is green and from 18th century.`,
        `An independent House, strategically located between Rembrand Square and National Opera, but where the bustle of the city comes to rest in this alley flowery and colorful.`,
      ],
      date: moment().format(`YYYY-MM-DD`),
    }],
  },

  {
    id: 2,
    title: `Beautiful & luxurious apartment at great location`,
    features: {
      type: `Apartment`,
      bedrooms: 3,
      adults: 4,
    },
    rating: 4.5,
    isPremium: true,
    price: 100,
    images: [
      `img/studio-01.jpg`,
      `img/apartment-01.jpg`,
      `img/apartment-02.jpg`,
      `img/apartment-03.jpg`,
      `img/studio-01.jpg`,
      `img/apartment-01.jpg`
    ],
    inside: [
      `Wi-Fi`,
      `Washing machine`,
      `Towels`,
      `Heating`,
      `Coffee machine`,
      `Baby seat`,
      `Kitchen`,
      `Dishwasher`,
      `Cabel TV`,
      `Fridge`,
    ],
    host: {
      name: `Angelina`,
      avatar: `img/avatar-angelina.jpg`,
      text: [
        `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam. The building is green and from 18th century.`,
        `An independent House, strategically located between Rembrand Square and National Opera, but where the bustle of the city comes to rest in this alley flowery and colorful.`,
      ]
    },

    coords: [52.3909553943508, 4.929309666406198],

    reviews: [{
      id: 2,
      user: {
        name: `Max`,
        avatar: `img/avatar-max.jpg`
      },
      rating: 4.5,
      text: [
        `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam. The building is green and from 18th century.`,
        `An independent House, strategically located between Rembrand Square and National Opera, but where the bustle of the city comes to rest in this alley flowery and colorful.`,
      ],
      date: moment().format(`YYYY-MM-DD`),
    }],
  },

  {
    id: 3,
    title: `Beautiful & luxurious apartment at great location`,
    features: {
      type: `Apartment`,
      bedrooms: 3,
      adults: 4,
    },
    rating: 4.5,
    isPremium: true,
    price: 100,
    images: [
      `img/studio-01.jpg`,
      `img/apartment-01.jpg`,
      `img/apartment-02.jpg`,
      `img/apartment-03.jpg`,
      `img/studio-01.jpg`,
      `img/apartment-01.jpg`
    ],
    inside: [
      `Wi-Fi`,
      `Washing machine`,
      `Towels`,
      `Heating`,
      `Coffee machine`,
      `Baby seat`,
      `Kitchen`,
      `Dishwasher`,
      `Cabel TV`,
      `Fridge`,
    ],
    host: {
      name: `Angelina`,
      avatar: `img/avatar-angelina.jpg`,
      text: [
        `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam. The building is green and from 18th century.`,
        `An independent House, strategically located between Rembrand Square and National Opera, but where the bustle of the city comes to rest in this alley flowery and colorful.`,
      ]
    },

    coords: [52.3809553943508, 4.939309666406198],

    reviews: [{
      id: 3,
      user: {
        name: `Max`,
        avatar: `img/avatar-max.jpg`
      },
      rating: 4.5,
      text: [
        `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam. The building is green and from 18th century.`,
        `An independent House, strategically located between Rembrand Square and National Opera, but where the bustle of the city comes to rest in this alley flowery and colorful.`,
      ],
      date: moment().format(`YYYY-MM-DD`),
    }],
  },

];

export const offer = {
  id: 0,
  title: `Beautiful & luxurious apartment at great location`,
  features: {
    type: `Apartment`,
    bedrooms: 3,
    adults: 4,
  },
  rating: 4.5,
  isPremium: true,
  price: 100,
  images: [
    `img/studio-01.jpg`,
    `img/apartment-01.jpg`,
    `img/apartment-02.jpg`,
    `img/apartment-03.jpg`,
    `img/studio-01.jpg`,
    `img/apartment-01.jpg`
  ],
  inside: [
    `Wi-Fi`,
    `Washing machine`,
    `Towels`,
    `Heating`,
    `Coffee machine`,
    `Baby seat`,
    `Kitchen`,
    `Dishwasher`,
    `Cabel TV`,
    `Fridge`,
  ],
  host: {
    name: `Angelina`,
    avatar: `img/avatar-angelina.jpg`,
    text: [
      `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam. The building is green and from 18th century.`,
      `An independent House, strategically located between Rembrand Square and National Opera, but where the bustle of the city comes to rest in this alley flowery and colorful.`,
    ]
  },

  coords: [52.3809553943508, 4.939309666406198],

  reviews: {
    id: 0,
    user: {
      name: `Max`,
      avatar: `img/avatar-max.jpg`
    },
    rating: 4.5,
    text: [
      `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam. The building is green and from 18th century.`,
      `An independent House, strategically located between Rembrand Square and National Opera, but where the bustle of the city comes to rest in this alley flowery and colorful.`,
    ],
    date: moment().format(`YYYY-MM-DD`),
  },
};
