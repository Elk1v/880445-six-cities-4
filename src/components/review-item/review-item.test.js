import React from "react";
import renderer from "react-test-renderer";
import ReviewITem from "ReviewItem/review-item.jsx";
import {review} from "Mocks/test-mocks";

it(`ReviewItem component renders correctly`, () => {
  const tree = renderer.create(
      <ReviewITem
        review={review}
      />
  ).toJSON();

  expect(tree).toMatchSnapshot();
});
