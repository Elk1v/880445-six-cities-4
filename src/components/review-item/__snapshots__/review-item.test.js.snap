// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ReviewItem component renders correctly 1`] = `
<li
  className="reviews__item"
>
  <div
    className="reviews__user user"
  >
    <div
      className="reviews__avatar-wrapper user__avatar-wrapper"
    >
      <img
        alt="Reviews avatar"
        className="reviews__avatar user__avatar"
        height={54}
        src="img/avatar-max.jpg"
        width={54}
      />
    </div>
    <span
      className="reviews__user-name"
    >
      Max
    </span>
  </div>
  <div
    className="reviews__info"
  >
    <div
      className="reviews__rating rating"
    >
      <div
        className="reviews__stars rating__stars"
      >
        <span
          style={
            Object {
              "width": "90%",
            }
          }
        />
        <span
          className="visually-hidden"
        >
          Rating
        </span>
      </div>
    </div>
    <p
      className="reviews__text"
    >
      A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam. The building is green and from 18th century.
    </p>
    <p
      className="reviews__text"
    >
      An independent House, strategically located between Rembrand Square and National Opera, but where the bustle of the city comes to rest in this alley flowery and colorful.
    </p>
    <time
      className="reviews__time"
      dateTime="2020-07-24"
    >
      July 2020
    </time>
  </div>
</li>
`;
